web/src/timerender.ts

```diff
@@ -243,6 +243,59 @@ export function last_seen_status_from_date(
     );
 }
 
+// Current date is passed as an argument for unit testing
+export function last_event_from_date(last_active_date: Date, current_date = new Date()): string {
+    const minutes = differenceInMinutes(current_date, last_active_date);
+    if (minutes <= 2) {
+        return $t({defaultMessage: "Just now"});
+    }
+    if (minutes < 60) {
+        return $t({defaultMessage: "{minutes} minutes ago"}, {minutes});
+    }
+
+    const days_old = differenceInCalendarDays(current_date, last_active_date);
+    const hours = Math.floor(minutes / 60);
+
+    if (hours < 24) {
+        if (hours === 1) {
+            return $t({defaultMessage: "An hour ago"});
+        }
+        return $t({defaultMessage: "{hours} hours ago"}, {hours});
+    }
+
+    if (days_old === 1) {
+        return $t({defaultMessage: "Yesterday"});
+    }
+
+    if (days_old < 90) {
+        return $t({defaultMessage: "{days_old} days ago"}, {days_old});
+    } else if (
+        days_old > 90 &&
+        days_old < 365 &&
+        last_active_date.getFullYear() === current_date.getFullYear()
+    ) {
+        // Online more than 90 days ago, in the same year
+        return $t(
+            {defaultMessage: "{last_active_date}"},
+            {
+                last_active_date: get_localized_date_or_time_for_format(
+                    last_active_date,
+                    "dayofyear",
+                ),
+            },
+        );
+    }
+    return $t(
+        {defaultMessage: "{last_active_date}"},
```

Comment:
These last two translation tags are useless -- you're asking the translator to translator a string that just contains a variable to be substituted.

Reply:
Ah that's interesting, it has been there for quite some time so I thought it was more so to translate the date into the user's locale date? I guess that's actually what `get_localized_date_or_time_for_format` does.